# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/performance/PerformanceWorker.h
# Commit: 1580a4ea1a85
# Full Hash: 1580a4ea1a8554df960c23f7d05fe1b1d8d4d7ef
# Author: Jens Stutte <jstutte@mozilla.com>
# Date: 2021-12-15 21:51:13
# Regressor Bug: 1744025
# File Overlap Count: 1
# Description:
#   Bug 1744025: Use CheckedUnsafePtr<WorkerPrivate> in PerformanceWorker. r=dom-worker-reviewers,edenchuang
#   
#   Depends on D132797
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132799
# ==============================================================================

diff -r af171636a87f -r 1580a4ea1a85 dom/performance/PerformanceWorker.h
--- a/dom/performance/PerformanceWorker.h	Wed Dec 15 13:48:18 2021 +0000
+++ b/dom/performance/PerformanceWorker.h	Wed Dec 15 13:48:19 2021 +0000
@@ -8,12 +8,11 @@
 #define mozilla_dom_PerformanceWorker_h
 
 #include "Performance.h"
+#include "mozilla/dom/WorkerPrivate.h"
 
 namespace mozilla {
 namespace dom {
 
-class WorkerPrivate;
-
 class PerformanceWorker final : public Performance {
  public:
   explicit PerformanceWorker(WorkerPrivate* aWorkerPrivate);
@@ -94,7 +93,7 @@
   }
 
  private:
-  WorkerPrivate* mWorkerPrivate;
+  CheckedUnsafePtr<WorkerPrivate> mWorkerPrivate;
 };
 
 }  // namespace dom
